lib/client/ipam_block.go

```diff
@@ -77,9 +77,9 @@ func (b *allocationBlock) autoAssign(
 
 	// Determine if we need to check for affinity.
 	checkAffinity := b.StrictAffinity || affinityCheck
-	if checkAffinity && b.HostAffinity != nil && host != *b.HostAffinity {
+	if checkAffinity && b.Affinity != nil && host != *b.Affinity {
```

Comment:
This isn't going to work.

`host` is still going to be of the form `hostname`, and `b.Affinity` is going to be of the form `host: hostname`

You probably want a small helper function for all of these cases.  e.g

```
// hostAffinityMatches returns true if the provided host matches the provided affinity.
func hostAffinityMatches(host, affinity) bool {
}
```